source/patterns/@aws-solutions-constructs/core/lib/dynamodb-table-helper.ts

```diff
@@ -60,6 +76,18 @@ export function buildDynamoDBTable(scope: Construct, props: BuildDynamoDBTablePr
   }
 }
 
+export function buildDynamoDBTableInterface(scope: Construct, props: BuildDynamoDBTableInterfaceProps): [dynamodb.ITable, dynamodb.Table?] {
```

Comment:
This can't be an additional function, we need consistency. The fact that it's called buildDynamoDBTableInterface is a clue there's a problem, because it builds a table - it ACCEPTS and interface. So it's still buildDynamoDBTable, and that tells us we need to change the other one, not write a new one. We need to change it in a way that minimizes impact on other clients.

Fortunately there are only 2 clients - lambda-dynamodb and apigateway-dynamodb. 

Change the existing routine to accept an existing table or and existing interface. If you get an inteface, return that interface and undefined.. If you get a table, return the table and the table. Otherwise create a table and return that table and that table.

Your call can remain unchanged. The existing calls can be either:

[this.dynamoTable, const ignoreInterface] = defaults.buildDynamoDBTable(this, props);

// or

this.dynamoTable = defaults.buildDynamoDBTable(this, props)[0];

In the near future we can discuss whether the existing constructs can be updated to work with interfaces.